apriltag_ros/src/common_functions.cpp

```diff
@@ -435,19 +444,10 @@ void TagDetector::addImagePoints (
     apriltag_detection_t *detection,
     std::vector<cv::Point2d >& imagePoints) const
 {
-  // Add to image point vector the tag corners in the image frame
-  // Going counterclockwise starting from the bottom left corner
-  double tag_x[4] = {-1,1,1,-1};
-  double tag_y[4] = {1,1,-1,-1}; // Negated because AprilTag tag local
-                                 // frame has y-axis pointing DOWN
-                                 // while we use the tag local frame
-                                 // with y-axis pointing UP
-  for (int i=0; i<4; i++)
+  //population of intrinsic-free corner pixel coordinates 
+  for(int i=3;i!=-1;i--) //clockwise from top left
   {
-    // Homography projection taking tag local frame coordinates to image pixels
-    double im_x, im_y;
-    homography_project(detection->H, tag_x[i], tag_y[i], &im_x, &im_y);
-    imagePoints.push_back(cv::Point2d(im_x, im_y));
+      imagePoints.push_back(cv::Point2d(detection->p[i][0],detection->p[i][1]));
```

Comment:
I'm not seeing any code that populates the ROS message with the pixel corner values.  Maybe you forgot to push those changes?

Reply:
So what I noticed was the function detectTags() returns an array of the message type AprilTagDetection. So at line 314, I populated tag_detection which is of the msg type, and this return is passed to some other function that publishes it. The higher up the hierarchy I go, the harder it is to follow what is being done as the top level structure isn't as straightforward to me as it would be to you. I'm assuming by your other comments that you acknowledged that this was what was done as you scrolled to my other changes further down.